File: TDD Boot Camp/Part 01/Code/OOPrinciples/src/com/oop/abstraction/Person.java
package com.oop.abstraction;

import java.time.Duration;
import java.time.LocalDate;

public class Person {
	public String firstName;
	public String lastName;
	public LocalDate  dateOfBirth;
	
	public int getAge() {
		return calculateAge();
	}
	
	int calculateAge() {
		long duration =  Duration.between(LocalDate.now(), this.dateOfBirth).getSeconds();
		return (int)(duration/(60*60*24*365));
	}
}

File: TDD Boot Camp/Part 01/Code/C#/OOPrinciples/Abstraction/Person.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Abstraction
{
    public class Person
    {
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public DateTime DateOfBirth { get; set; }

        public int GetAge()
        {
            return CalculateAge();
        }

        int CalculateAge()
        {
            double duration = (DateTime.Now - DateOfBirth).TotalSeconds;
            return (int)(duration / (60 * 60 * 24 * 365));
        }
    }
}
